waspc/src/Wasp/Generator/DbGenerator/Auth.hs

```diff
@@ -0,0 +1,121 @@
+module Wasp.Generator.DbGenerator.Auth where
+
+import Data.Function ((&))
+import Data.Maybe
+import qualified Data.Text as T
+import NeatInterpolation (trimming)
+import Wasp.Analyzer.StdTypeDefinitions.Entity (parsePslBody)
+import qualified Wasp.AppSpec.Entity as AS.Entity
+import Wasp.Generator.Monad
+  ( Generator,
+    GeneratorError (GenericGeneratorError),
+    logAndThrowGeneratorError,
+  )
+import qualified Wasp.Psl.Ast.Model as Psl.Model
+import qualified Wasp.Psl.Ast.Model as Psl.Model.Field
+
+authEntityName :: String
+authEntityName = "Auth"
+
+authEntityIdType :: String
+authEntityIdType = "String"
+
+userFieldOnAuthEntityName :: String
+userFieldOnAuthEntityName = "user"
+
+authFieldOnUserEntityName :: String
+authFieldOnUserEntityName = "auth"
+
+authFieldOnProviderEntityName :: String
+authFieldOnProviderEntityName = "auth"
+
+providerEntityName :: String
+providerEntityName = "SocialAuthProvider"
```

Comment:
I wonder if we should name it precisely, since it is a specific type of Provider -> Oauth2, right? So maybe we should call it `OAuth2Provider` instead? Actually word `Social` probably doesn't have the place here, since it is not just facebook and LinkedIn auth -> we probably wouldn't call Github or Google as social auths. Or would we? Anyway I would suggest making sure we figure out the best name for this one.

Reply:
The name was changed to `AuthIdentity` as we agreed that it captures better what it is.